### PR TITLE
Add missing sitl setup and packages up to

### PR DESCRIPTION
**Pull request by Ryanf55**
This improves instructions to make sure users don't miss steps for SITL.

It also improves the build workflow by only building what's needed, instead of everything, which is good when someone's ROS workspace is huge. 

This work was sponsored by AeroVironment.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
if this is ready to merge let me know

File: dev/source/docs/ros2-sitl.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ROS 2 with SITL
 ===============
 
-Once ROS2 is correctly :ref:`installed <ros2>`, source your workspace and launch Ardupilot SITL with ROS 2! 
+Once ROS2 is correctly :ref:`installed <ros2>`, and SITL is also :ref:`installed <sitl-simulator-software-in-the-loop>`, source your workspace and launch Ardupilot SITL with ROS 2!
```
Comment (srmainwaring) [inline dev/source/docs/ros2-sitl.rst:7]: All looks fine. From the discord discussions with users the main thing I'd want to emphasise is that users should have a working ArduPilot SITL environment (which requires all deps be installed) before attempting to build and run with ROS 2 integration.

**Review by srmainwaring — APPROVED** [review]

File: dev/source/docs/ros2-sitl.rst (hunk 1 above)
Comment (Ryanf55) [reply dev/source/docs/ros2-sitl.rst:7]: Should I change the wording here to be `installed and working` instead of `installed`?

**Comment by Ryanf55** [comment]
> if this is ready to merge let me know

I'm running the wiki build now to confirm rendering looks good. I'll tag you when everything is good.

**Comment by Ryanf55** [comment]
Sorry, after running through all the steps on a fresh machine, the build failed. I'll have to debug this next week at work. We might be missing a step to explicitely say to source the ArduPilot virtual environment at some point. 
```
rfriedman@dev-rfriedman01:~/Dev/ardu_ws$ colcon build --packages-up-to ardupilot_dds_tests --cmake-args -DBUILD_TESTING=ON
Starting >>> micro_ros_agent
Finished <<< micro_ros_agent [0.62s]                     
Starting >>> ardupilot_sitl
--- stderr: ardupilot_sitl                               
openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

openjdk version "11.0.20.1" 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)

Build failed
Traceback (most recent call last):
  File "/home/rfriedman/Dev/ardu_ws/src/ardupilot/modules/waf/waflib/Task.py", line 348, in process
    ret = self.run()
  File "/home/rfriedman/Dev/ardu_ws/src/ardupilot/Tools/ardupilotwaf/mavgen.py", line 54, in run
    from pymavlink.generator import mavgen
  File "/home/rfriedman/Dev/ardu_ws/src/ardupilot/modules/mavlink/pymavlink/generator/mavgen.py", line 26, in <module>
    from future import standard_library
ModuleNotFoundError: No module named 'future'

gmake[2]: *** [CMakeFiles/ardupilot_build.dir/build.make:70: CMakeFiles/ardupilot_build] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:168: CMakeFiles/ardupilot_build.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< ardupilot_sitl [18.5s, exited with code 2]
```

**Comment by Ryanf55** [comment]
I have it figured out now; I needed to install the ArduPilot toolchains and have added that as a pre-req.

The next error is unrelated, and is a problem in the repo, not the wiki.
```
- ardupilot_dds_tests.test.ardupilot_dds_tests.test_virtual_ports test_virtual_ports
  <<< failure message
    AssertionError: Test process had no output.
    assert 'N starting data transfer loop' in '/bin/sh: 1: socat: not found\n'
  >>>

```

**Comment by srmainwaring** [comment]
> socat: not found

Missing a `sudo apt install socat`.

We also use `ros-humble-launch-pytest` which is not installed as part of desktop-full (but is available for Humble).

**Review by srmainwaring — APPROVED** [review]
Is reading much better thanks @Ryanf55.

**Comment by Ryanf55** [comment]
> if this is ready to merge let me know

Ready for merge!

**Pull request merged**